File: palindrome.py
def isPalindrome(userInp):                                                  // return type does not need to be specified in python. parameter does not need type either
    for i in range(0, int(len(userInp)/2)):                                 // iterate i from 0 to half of the size of the inp string
        if userInp[i] != userInp[len(userInp)-1-i]:                         // if characters do not match, return false
            return False
    return True                                                             // all characters match, inp is a palindrome, return true

userInput = input("Enter a word to check if it is a palindrome: ")          // type is not specified when userInput is declared. userInput is declared, the string is outputted to the console and the user input is assigned to userInput all in one line

flag = isPalindrome(userInput);                                             // type is not specified when flag is declared. flag is set to the return value of isPalindrome

if flag:                                                                    // if flag is true
    print(userInput + " is a palindrome")
else:                                                                       // if flag is false
    print(userInput + " is not a palindrome")

File: palindrome.cpp
#include <iostream>                                                 // mandatory file for input/output

using namespace std;                                                // for standard namespace

bool isPalindrome(string inp){                                      // return type has to be explicitly specified, parameter type also has to be specified
    for( int i = 0; i < inp.size()/2; ++i ){                        // iterate i from 0 to half of the size of the inp string
        if ( inp[i] != inp[inp.size()-1-i] ) {                      // if characters do not match, return false
            return false;
        }
    }
    return true;                                                    // all characters match, inp is a palindrome, return true
}

int main()                                                          // required main function that all c++ programs need
{
    string userInput;                                               // data type 'string' required in c++
    cout << "Enter a word to check if it is a palindrome: ";        // output string to console
    cin >> userInput;                                               // get user input from console and store in userInput
    
    bool flag = isPalindrome(userInput);                            // data type 'bool' required, store result of function return value in flag (true if userInput is a palindrome, false otherwise)
    
    if (flag) {                                                     // if flag is true
        cout << userInput << " is a palindrome" << endl;
    } else {                                                        // if flag is false
        cout << userInput << " is not a palindrome" << endl;
    }
    
    return 0;                                                       // return 0 to end main function
}
